### PR TITLE
1657389 Add descriptions for SEO (series of related PRs)

### PR DESCRIPTION
**Pull request by TimShererWithAquent**
## Add descriptions for SEO

For anyone that GitHub notifies, this is for [1657389](https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1657389).

You're welcome but not required, according to KhairunJ's guidance, to review these proposed meta descriptions. I'll leave this PR open for 48 hours and address any feedback. After that, we'll sign off for merge. Thanks!

Please keep in mind our 100-160 char length for descriptions.

Fixes   #1657389

### CODE REVIEW

**Comment by TimShererWithAquent** [comment]
Closing and reopening to try to clear that failed check. Not familiar with that, but maybe a restart will clear it.

**Comment by TimShererWithAquent** [comment]
And reopen.

File: docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Named and Optional Arguments - C# Programming Guide"
+description: Named arguments in C# specify arguments by name, not position. Optional arguments can be omitted. Named & optional arguments facilitate calls to COM interfaces.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/named-and-optional-arguments.md:3]: Let's drop the last sentence. Named and optional arguments will be much more useful for records (coming in C# 9)
```suggestion
description: Named arguments in C# specify arguments by name, not position. Optional arguments can be omitted.
```

File: docs/csharp/programming-guide/classes-and-structs/nested-types.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Nested Types - C# Programming Guide"
+description: A type defined within a class, struct, or interface is called a nested type in C#. Nested types are accessible only from their containing type.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/nested-types.md:3]: The last sentence is incorrect. (If they have the default, `private`, access, it's true.)

```suggestion
description: A type defined within a class, struct, or interface is called a nested type in C#. 
```

File: docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Object and Collection Initializers - C# Programming Guide"
+description: Object initializers in C# assign values to accessible fields or properties of an object at creation without invoking a constructor and assignment statements.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md:3]: In this context, object initializers are executed *after* the constructor:

```suggestion
description: Object initializers in C# assign values to accessible fields or properties of an object at creation after invoking a constructor.
```

File: docs/csharp/programming-guide/classes-and-structs/objects.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Objects - C# Programming Guide"
+description: C# uses a class or struct definition to create an object. In an object-oriented language such as C#, a program consists of objects interacting dynamically.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/objects.md:3]: Definitions don't create an object:

```suggestion
description: C# uses a class or struct definition to define types of objects. In an object-oriented language such as C#, a program consists of objects interacting dynamically.
```

File: docs/csharp/programming-guide/classes-and-structs/private-constructors.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ---
 title: "Private Constructors - C# Programming Guide"
+description: A private constructor is a special instance constructor in C#, generally used in classes that contain static members only.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/private-constructors.md:3]: There are other good uses for private constructors. (factory methods, etc). Let's leave it open in the description.

```suggestion
description: A private constructor is a special instance constructor in C# used to restrict how an object can be created. They may be used with factory methods or other construction idioms.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks @TimShererWithAquent 

I had a few comments on some of the descriptions. Once those are addressed, I'll :shipit:

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @TimShererWithAquent 

FYI: From the "files" tab, you can add all the suggestions to a batch and add them in one commit.

**Pull request merged**